cmd/kubectl/app/plugins.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"os"
+	"strings"
+
+	cmdplugin "k8s.io/kubernetes/pkg/kubectl/cmd/plugin"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/plugin/mesosphere"
+)
+
+var pluginRegistrationFuncs = []func() error{
+	// add all plugin registation funcs here
+	mesosphere.RegisterPlugins,
+}
+
+func RegisterPlugins() error {
+	for _, f := range pluginRegistrationFuncs {
+		if err := f(); err != nil {
+			return err
+		}
+	}
+	pluginEnv := os.Getenv("KUBECTL_PLUGINS")
```

Comment:
implementing this outside of config gives a plugin the change to alter the config loading process/rules. specifically w/ respect to flag processing. if config plugin activation was driven by a config file ... it would seem to further complicate the loading rules for config processing. for example, if I wanted to add a `--dcos-token` flag override how would I do that? require another pass over the config file(s) after the plugins have been activated?